### PR TITLE
[Finishes #187355051] user should be able to login using email and password

### PR DESCRIPTION
**Pull request by Celestin4**
**1. What does this PR do?**

This is a feature that allow user to  on login our website using email and password. 

**2. Description of task to be completed**

Why
To be able to use the application and access authenticated routes, a login form should be presented to users.

Acceptance Criteria

GIVEN user successfully logs in.
WHEN user passes appropriate login parameters
THEN user should be able to login.

DEV NOTE (REQUIRED)

Under activity, endeavor to come up with the thought process (data flow) needed to implement this feature.
Create a breakdown of steps as tasks under the task attribute below of PT.
The login in form should present to the user descriptive validation errors incase of any.
The Login page should present a link for users to Reset their passwords incase of forgotten credentials
Kindly do not user Third party packages and or Libraries such as ReduxForm. Use plain html ,CSS Forms with any front end framework such as Bootstrap, MaterializeCSS etc

**3. How should this be manually tested?**

Clone the repository.
Checkout to the branch.
Install necessary dependencies.
Start the development server.

****

### CODE REVIEW

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good work @Celestin4 but please consider the following:
1. Add a hide/show password option
Screenshot 2024-06-24 at 18 39 16

2. Add a valid backend link so that we can test your feature
Screenshot 2024-06-24 at 18 38 33

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
@Celestin4, I can not test your PR, please check your backend URL:
Screenshot 2024-06-26 at 08 54 05

**Review by lilika67 — APPROVED** [review]

**Review by Christian-Ishimwe — APPROVED** [review]

**Review by shema-surge — APPROVED** [review]

**Review by pacifiquemboni — APPROVED** [review]

**Comment by UmuhireJessie** [comment]
Can you please add deployment for your branch?

File: src/redux/slices/loginSlice.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    error: any;
+    isSucceeded: boolean;
+}
+
+export const loginUser = createAsyncThunk<UserData, UserData, { rejectValue: any }>(
+    'login/loginUser',
+    async (userData, { rejectWithValue }) => {
+        try {
+            const response = await axios.post(`${BACKEND_URL}/api/users/login`, userData);
+            localStorage.setItem('token', response.data.token);
```
Comment (UmuhireJessie) [inline src/redux/slices/loginSlice.ts:28]: It is advisable to hash the token so everyone in production can't see it. But this is okay for now

**Review by UmuhireJessie — APPROVED** [review]

**Comment by UmuhireJessie** [comment]
It is advisable to hash a JWT on the front end is advisable so that just everyone does not see it. Also, please change your commit message

**Pull request merged**